stub-runner/stub-runner/src/main/groovy/com/ofg/stub/registry/StubRegistry.groovy

```diff
@@ -54,7 +55,26 @@ class StubRegistry {
                 .uriSpec(URI_SPEC)
                 .address('localhost')
                 .port(stubServer.port)
-                .name(stubServer.projectMetadata.projectRelativePath)
+                .name(serviceName(stubServer))
                 .build()
     }
+
+    private static String serviceName(StubServer stubServer) {
+        String projectRelativePath = stubServer.projectMetadata.projectRelativePath
+        String projectName = stubServer.projectMetadata.projectName
+        if (projectRelativePath.endsWith(projectName)) {
+            return projectRelativePath
+        } else {
+            return pathWithoutProjectName(projectRelativePath) + projectName
+        }
+    }
+
+    private static String pathWithoutProjectName(String projectRelativePath) {
+        if (projectRelativePath.contains('/')) {
+            int idx = projectRelativePath.lastIndexOf('/')
+            return projectRelativePath.substring(0, idx + 1)
```

Comment:
Not really. There could more than one `/` in `projectRelativePath` and we need whole path till the last occurrence of `/`.